plugins/org.locationtech.udig.project/src/org/locationtech/udig/project/internal/commands/edit/SetAttributesCommand.java

```diff
@@ -60,6 +60,7 @@ public SetAttributesCommand( IBlockingProvider<SimpleFeature> feature, IBlocking
             Object value[] ) {
         this.xpath = xpath;
         this.value = value;
+        this.oldValue = new Object[value.length];
```

Comment:
Maybe its worth to check for null of value here. Looks like the only caller is EditFeature where value is set. However this is an implementation detail but from javadoc constructor argument description it could be possible if somebody like to implement to "clean"/"empty" attributes.

I would suggest to check for null in both constructors or even delegate from the second to the first one using:
```
public SetAttributesCommand( String xpath[], Object value[] ) {
        this(new EditFeatureProvider(this), new EditLayerProvider(this), xpath, value); 
}
```

Reply:
Hi @fgdrf, 
Delegation the way you specify it is not possible since  we cannot refer to 'this' while explicitly invoking a constructor. 
Regarding null value check what can be done is to put an Assert.isNotNull for both xpath[] and value[] at the beginning of each constructor, since providing null values for these parameters is more or less meaningless for this specific command and should be not permitted (the subsequent execution of the run method will fail). 
If a user wants to clean/empty attributes he should set a value[] array with the proper length and null values for each array element desired to be cleaned 
If this will this suffice your request I will add the Assertion checks.

Reply:
> Delegation the way you specify it is not possible since we cannot refer to 'this' while explicitly invoking a constructor.

Oh yes, haven't seen this. However, the first Constructor is never used in codebase. I'll provide a pull request to eliminate it.

I'm wondering a bit why value.length is used while the loop 
```
		// for loop to get the old values
		for(int i = 0; i < xpath.length; i++){
		    oldValue[i] = feature2.getAttribute(xpath[i]);
		}
```

uses `xpath.lenght`. I assume that object-Array might have same length but who knows ;)

Reply:
see pull https://github.com/nprigour/udig-platform/pull/10 for validation. I would merge afterwards..

Reply:
Going to change the pull ;)